gobblin-service/src/main/java/org/apache/gobblin/service/modules/core/GobblinServiceGuiceModule.java

```diff
@@ -0,0 +1,253 @@
+import org.apache.gobblin.service.modules.topology.TopologySpecFactory;
+import org.apache.gobblin.service.modules.utils.HelixUtils;
+import org.apache.gobblin.service.modules.utils.InjectionNames;
+import org.apache.gobblin.service.monitoring.FlowStatusGenerator;
+import org.apache.gobblin.service.monitoring.FsJobStatusRetriever;
+import org.apache.gobblin.service.monitoring.JobStatusRetriever;
+import org.apache.gobblin.service.monitoring.KafkaJobStatusMonitor;
+import org.apache.gobblin.service.monitoring.KafkaJobStatusMonitorFactory;
+import org.apache.gobblin.util.ClassAliasResolver;
+import org.apache.gobblin.util.ConfigUtils;
+
+
+public class GobblinServiceGuiceModule implements Module {
+  private static final Logger LOGGER = LoggerFactory.getLogger(GobblinServiceGuiceModule.class);
+  private static final String JOB_STATUS_RETRIEVER_CLASS_KEY = "jobStatusRetriever.class";
+
+  GobblinServiceConfiguration serviceConfig;
+
+  public GobblinServiceGuiceModule(GobblinServiceConfiguration serviceConfig) {
+    this.serviceConfig = Objects.requireNonNull(serviceConfig);
+  }
+
+  @Override
+  public void configure(Binder binder) {
+    LOGGER.info("Configuring bindings for the following service settings: {}", serviceConfig);
+
+    // In the current code base, we frequently inject classes instead of interfaces
+    // As a result, even when the binding is missing, Guice will create an instance of the
+    // the class and inject it. This interferes with disabling of different services and
+    // components, because without explicit bindings they will get instantiated anyway.
+    binder.requireExplicitBindings();
+
+    // Optional binder will find the existing binding for T and create additional binding for Optional<T>.
+    // If none of the specific class binding exist, optional will be "absent".
+    OptionalBinder.newOptionalBinder(binder, Logger.class);
+
+    binder.bind(Logger.class).toInstance(LoggerFactory.getLogger(GobblinServiceManager.class));
+
+    binder.bind(Config.class).toInstance(serviceConfig.getInnerConfig());
+
+    binder.bind(GobblinServiceConfiguration.class).toInstance(serviceConfig);
+
+    // Used by TopologyCatalog and FlowCatalog
+    GobblinInstanceEnvironment gobblinInstanceEnvironment = StandardGobblinInstanceLauncher.builder()
+        .withLog(LoggerFactory.getLogger(GobblinServiceManager.class))
+        .setInstrumentationEnabled(true)
+        .withSysConfig(serviceConfig.getInnerConfig())
+        .build();
+
+    binder.bind(GobblinInstanceEnvironment.class).toInstance(gobblinInstanceEnvironment);
+
+    binder.bind(EventBus.class)
+        .annotatedWith(Names.named(GobblinServiceManager.SERVICE_EVENT_BUS_NAME))
+        .toInstance(new EventBus(GobblinServiceManager.class.getSimpleName()));
+
+    binder.bindConstant().annotatedWith(Names.named(InjectionNames.SERVICE_NAME)).to(serviceConfig.getServiceName());
+
+    binder.bindConstant()
```

Comment:
Is the general recommendation then that all service wide constants which were getting passed around in config objects or as arguments use DI instead?

Reply:
If the class has many configuration settings, than it's better to have a config object and inject it. That's what we do for GobblinServiceManager that gets GobblinServiceConfiguration injected into it.

Service name is injected into a couple of classes that don't have much in common, and those classes don't have a lot of configuration of their own. In future PRs we can consider adding something like ServiceEnvironment class that will hold the service name, id and other properties that can be useful for the classes that run inside Gobblin service. 

There are still a bunch of places in the code that needs to be refactored for better DI support, but I left them out at the moment to reduce the PR size.